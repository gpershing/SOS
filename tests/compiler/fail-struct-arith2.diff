diff: ./tests/compiler/tests/compiler/fail-struct-arith2.err: No such file or directory
