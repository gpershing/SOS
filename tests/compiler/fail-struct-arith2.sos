struct s1 = {int a, float  b}

struct s2 = {int a, float b}

s1 + s2
