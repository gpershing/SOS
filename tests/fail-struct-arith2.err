Fatal error: exception Failure("Can only add or subtract structs of matching type")
