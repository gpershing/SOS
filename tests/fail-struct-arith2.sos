//arithmetic on structs that do not match types

struct s1 = {a: int, b: float}

struct s2 = {a: float, b: int}

a: s1 = {1, 1.0}
b: s2 = {1.0, 1}

a+b

