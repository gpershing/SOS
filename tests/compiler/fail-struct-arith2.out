Can only operate on arithmetic structs
